src/accountsdb/accounts_file.zig

```diff
@@ -10,7 +10,27 @@ const Pubkey = @import("../core/pubkey.zig").Pubkey;
 
 const AccountFileInfo = @import("snapshots.zig").AccountFileInfo;
 
-pub const FileId = u32;
+/// Simple strictly-typed alias for an integer, used to represent a file ID.
+pub const FileId = enum(u32) {
+    _,
+
+    pub inline fn fromInt(int: u32) FileId {
```

Comment:
I am often tempted to inline functions when it seems like a good fit (as in this case). I usually choose not to because I've heard it suggested many times by people more knowledgeable than me about compilers that the compiler is better at making these decisions. I'm curious about your thoughts.

Example: https://ziggit.dev/t/inlining-functions/1341/3

Reply:
My reasoning here is that these functions ought to compile down to their exact function bodies; there are no smart optimizations the compiler can make here beyond simply doing exactly what I've already done, which is to inline the function into the builtin call. This work could be left to the optimizer, but then that requires that the optimizer commit to more work than is strictly necessary. I already know there is nothing else it can do, so by marking it inline I am giving it one less thing to do.

This isn't performance critical, and it won't come up in any measurements, but it expresses my intent for these functions to be nothing more than direct wrappers over these builtins. If they became more complicated, or I suspected the compiler could intuit more than me, I would then also heed Andrew's advice.

As well, this will not have any effect on any stack traces, because these functions cannot panic or return an error, and thus their inline-ness will have no impact on the stack frames.

Lmk what you think. I am open to un-inlining these if we want to adhere to a strict policy of only `inline`ing when strictly necessary or measurably beneficial.

Reply:
I'm not convinced that we need to strictly enforce a policy like that. And I don't see any harm of inlining this function. Mainly I just wanted to hear your thoughts. I'm fine with keeping it as is.